### ISSUE TITLE
Found with fuzzer: bad bag o' tricks

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#42]
```
Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
#2  0x0000555555a4d4df in NH_abort () at end.c:236
#3  0x0000555555a54335 in panic (str=0x5555565429e0 "%s") at end.c:808
#4  0x0000555555fac0a8 in impossible (s=0x5555564db580 "bad bag o' tricks") at pline.c:518
#5  0x0000555555b74dc4 in bagotricks (bag=0x60c0002f3e00) at makemon.c:4566
#6  0x0000555555fa7dad in tipcontainer_checks (box=0x60c0002f3e00, targetbox=0x0, 
    allowempty=0 '\000') at pickup.c:3942
#7  0x0000555555fa5782 in tipcontainer (box=0x60c0002f3e00) at pickup.c:3656
#8  0x0000555555fa2691 in dotip () at pickup.c:3425
#9  0x0000555555868d6e in rhack (cmd=0x5555567c9f40 <in_line> <incomplete sequence \324>)
    at cmd.c:5531
#10 0x0000555555767f7b in moveloop (resuming=0 '\000') at allmain.c:823
#11 0x00005555563d7cc9 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737335783280, 93824997446065, 335544320, 
            140737341046928, 1073741824, 0, 140737342793376, 4294967295, 140737337360384, 
            140737342776480, 2899516599963546624}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342776480, 2899516599963546624, 
              140737488343024, 140737342793376, 18446744073709547520, 140737327104000, 
              140737345462488, 140737488344096, 140737488344096, 140737344331794, 
              17592186042878, 140737488338944, 140737341046928, 2899516599963546624, 
              140737488343424}}, sa_flags = 1093465088, sa_restorer = 0x7fffffffd340}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a4d4df in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a54335 in panic (str=0x5555565429e0 "%s") at end.c:808
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffd410, 
            reg_save_area = 0x7fffffffd340}}
#4  0x0000555555fac0a8 in impossible (s=0x5555564db580 "bad bag o' tricks") at pline.c:518
        pbuf = "bad bag o' tricks\000\377\377\377\177\000\000\060\325\377\377\377\177\000\000\060\325\377\377\377\177\000\000k\330\377\377\377\177\000\000\260\326\377\377\377\177\000\000-\022@VUU\000\000\000\000\000\000\003\000\000\000\017\000\000\000\000\000\000\000\003\000\000\000\b", '\000' <repeats 15 times>, "\003\000\000\000\002\000\000\000\377\377\377\377\000\000\000\000\036", '\000' <repeats 11 times>, "\200I\b\000\020a\000\000\000\000\000\000\030\000\000\000\263\212\265A\000\000\000\000f\303]VUU\000\000\276\360?VUU\000\000\000\000\000\000\000\000\000\000\263\212\265A\000\000\000\000\224\305]VUU\000\000]\377?VUU", '\000' <repeats 26 times>...
        the_args = {{gp_offset = 8, fp_offset = 48, overflow_arg_area = 0x7fffffffdb00, 
            reg_save_area = 0x7fffffffda20}}
#5  0x0000555555b74dc4 in bagotricks (bag=0x60c0002f3e00) at makemon.c:4566
No locals.
#6  0x0000555555fa7dad in tipcontainer_checks (box=0x60c0002f3e00, targetbox=0x0, 
    allowempty=0 '\000') at pickup.c:3942
        maybeshopgoods = 0 '\000'
        oy = 4 '\004'
        res = 0
        bag = 1 '\001'
        old_spe = 10
        ox = 48 '0'
#7  0x0000555555fa5782 in tipcontainer (box=0x60c0002f3e00) at pickup.c:3656
        ox = 48 '0'
        oy = 4 '\004'
        held = 0 '\000'
        maybeshopgoods = 0 '\000'
        targetbox = 0x0
        cancelled = 0 '\000'
#8  0x0000555555fa2691 in dotip () at pickup.c:3425
        cobj = 0x60c0002f3e00
        nobj = 0x0
        cc = {x = 48 '0', y = 4 '\004'}
        boxes = 0
        c = 0 '\000'
        buf = '\000' <repeats 152 times>, "\340\270T\367\377\177\000\000\026\375=VUU", '\000' <repeats 89 times>
        qbuf = '\000' <repeats 105 times>, "\364,A\374&=(\000\000\000\000\000\000\000\000Y\375=VUU\000\000\000\000\000\000\377\177\000\000*/\177VUU\000\000\060\342\377\377\377\177\000\000\204\375=VUU\000\000\060\342\377\377\377\177\000\000\020\001\000\000\320`\000\000p\342\377\377\377\177\000\000\251\021?VUU\000\000\000\344\377\377\377\177\000\000\000\000\000\000"...
        spillage = 0x0
#9  0x0000555555868d6e in rhack (cmd=0x5555567c9f40 <in_line> <incomplete sequence \324>)
    at cmd.c:5531
        tlist = 0x555556687380 <extcmdlist+4032>
        res = 688
        func = 0x555555fa1779 <dotip>
        spkey = 0
        prefix_seen = 0 '\000'
        bad_command = -4 '\374'
        firsttime = 1 '\001'
#10 0x0000555555767f7b in moveloop (resuming=0 '\000') at allmain.c:823
        moveamt = 12
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 31062
        past_clock = 70766
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#11 0x00005555563d7cc9 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
Could not reproduce by tipping bag of tricks, bag of rats, large box, or regular sack.

**Comment by elunna** [comment]
Occurs when a horn of plenty is #tipped.

commit 815ca37c9: Fix for issue #42. dotip had an incorrect check for bag of tricks or rats.
